From bd95b681f5b0d3321de342dbe49610261402b024 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:27:41 -0400
Subject: [tools] perf/trace: Beautify fifth argument of mmap() as fd

Message-id: <1408462094-14194-293-git-send-email-jolsa@redhat.com>
Patchwork-id: 88309
O-Subject: [PATCH RHEL7.1 BZ1131394 292/325] perf trace: Beautify fifth argument of mmap() as fd
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 73faab3a421351c342b10b0177fec9eea1a1d809
Author: Namhyung Kim <namhyung.kim@lge.com>
Date: Tue Nov 12 15:24:59 2013 +0900

description
===========
The fifth argument of mmap syscall is fd and it often contains -1 as a
value for anon mappings.  Without this patch it doesn't show the file
name as well as it shows -1 as 4294967295.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-trace.c b/tools/perf/builtin-trace.c
index c3008b1..aeb6296 100644
--- a/tools/perf/builtin-trace.c
+++ b/tools/perf/builtin-trace.c
@@ -951,7 +951,8 @@ static struct syscall_fmt {
  { .name     = "mmap",     .hexret = true,
    .arg_scnprintf = { [0] = SCA_HEX,   /* addr */
         [2] = SCA_MMAP_PROT, /* prot */
-        [3] = SCA_MMAP_FLAGS, /* flags */ }, },
+        [3] = SCA_MMAP_FLAGS, /* flags */
+        [4] = SCA_FD,    /* fd */ }, },
  { .name     = "mprotect",   .errmsg = true,
    .arg_scnprintf = { [0] = SCA_HEX, /* start */
         [2] = SCA_MMAP_PROT, /* prot */ }, },
-- 
1.7.1